src/common/chess-engine.ts

```diff
@@ -0,0 +1,53 @@
+// engine.ts
+
+import { Chess, Square } from "chess.js";
+import { GameFinishedReason } from "./game-end";
+
+export class ChessEngine {
+    private chess: Chess;
+
+    constructor() {
+        this.chess = new Chess();
+    }
+
+    reset() {
+        this.chess.reset();
+    }
+
+    getMoves(passedSquare?: Square) {
+        return this.chess.moves({ square: passedSquare, verbose: true });
+    }
+
+    get fen() {
+        return this.chess.fen();
+    }
+
+    makeMove(from: string, to: string) {
```

Comment:
Type should be `Square` instead of `string`. We might also want to switch to our own custom `Square` type to improve consistency.